graph/push.go

```diff
@@ -478,6 +514,18 @@ func (s *TagStore) CmdPush(job *engine.Job) engine.Status {
 	if !exists {
 		return job.Errorf("Repository does not exist: %s", repoInfo.LocalName)
 	}
+
+	if endpoint.Version == registry.APIVersion2 {
```

Comment:
Originally there was additional logic for v1 push which checked for push by image ID. Since v2 does not support push by the random image ID and never will, the v2 code was branched before.  Since that code is now removed, this unifies at least the initial Repository exist check for v1 and v2.